frontend/pages/admin/UserManagementPage/components/UserForm/UserForm.tsx

```diff
@@ -140,6 +177,10 @@ const UserForm = ({
 
   const initiallyPasswordAuth = !isSsoEnabled;
 
+  useEffect(() => {
+    setFormErrors({ ...formErrors, ...ancestorErrors });
+  }, [ancestorErrors]);
```

Comment:
Intentionally not depending on `formErrors` so this effect only runs on update of ancestor errors, which will be any server errors returned by the API. All validation of current form values as the user interacts before submission is handled onChange and onBlur of each field.

Reply:
This fixes the bug as outlined by the issue. Rest of the changes are moving validation of this form towards our ideal as defined in `frontend/docs/patterns.md`, which also makes it easier to reason about, especially given the intricate interactions of data/handlers coming in as props from multiple parents and data managed by this component.